src/Structures.hs

```diff
@@ -89,7 +89,9 @@ instance Showable Bidding where
                 foldr foldFormatMaybeBid ([], nextFootnote) row
           in
             ((join " " . reverse $ rowBids) : results, nextFootnote')
-        formatAuction = unlines . reverse . fst .
+        -- `unlines` always puts a newline at the end. We remove it with `init`
+        -- so we can append the " ??" on the end of the final line.
+        formatAuction = init . unlines . reverse . fst .
```

Comment:
It's weird that this needed a newline character removed; I don't recall seeing that on my other computer. Check on this before merging. Maybe I should call `strip` instead of `init`?